File: Tablet.java
package SistemaGestionTienda;

public class Tablet extends ProductoElectronico {
    private boolean tieneLapiz;
    private String sistemaOperativo;

    // Constructor con sobrecarga
    public Tablet(String idProducto, String nombre, String marca, double precio,
                  int stock, boolean tieneLapiz, String sistemaOperativo) {
        super(idProducto, nombre, marca, precio, stock);
        this.tieneLapiz = tieneLapiz;
        this.sistemaOperativo = sistemaOperativo;
    }

    // Getters y Setters
    public boolean getTieneLapiz() {
        return tieneLapiz;
    }

    public String getSistemaOperativo() {
        return sistemaOperativo;
    }

    public void setTieneLapiz(boolean tieneLapiz) {
        this.tieneLapiz = tieneLapiz;
    }

    public void setSistemaOperativo(String sistemaOperativo) {
        this.sistemaOperativo = sistemaOperativo;
    }

    @Override
    public void mostrarInformacion() {
        System.out.println("Tablet: " + getNombre() +
                           " | Marca: " + getMarca() +
                           " | SO: " + sistemaOperativo +
                           " | ¿Tiene lápiz?: " + (tieneLapiz ? "Sí" : "No") +
                           " | Precio: €" + getPrecio() +
                           " | Stock: " + getStock());
    }

    @Override
    public void actualizarPrecio(double porcentaje) {
        double nuevoPrecio = getPrecio() * (1 + porcentaje / 100);
        setPrecio(nuevoPrecio);
    }
}

File: Tablet.py
from Producto_electronico import ProductoElectronico

class Tablet(ProductoElectronico):
    def __init__(self, id_producto, nombre, marca, precio, stock, tiene_lapiz, sistema_operativo):
        super().__init__(id_producto, nombre, marca, precio, stock)
        self.tiene_lapiz = tiene_lapiz
        self.sistema_operativo = sistema_operativo

    def mostrar_informacion(self):
        print(f"{self.nombre} ({self.marca}) - SO: {self.sistema_operativo} - Lápiz: {'Sí' if self.tiene_lapiz else 'No'} - Precio: €{self.precio:.2f} - Stock: {self.stock}")

    def actualizar_precio(self, porcentaje):
        self.precio *= (1 + porcentaje / 100)
